File: api/internal/pkg/upload/buckets.go
package upload

var Buckets = []string{
	"default",
	"profile-images",
	"banner-images",
	"pois",
	"reviews",
	"diaries",
}

type BucketName string

const (
	BUCKET_DEFAULT        BucketName = "default"
	BUCKET_PROFILE_IMAGES BucketName = "profile-images"
	BUCKET_BANNER_IMAGES  BucketName = "banner-images"
	BUCKET_POIS           BucketName = "pois"
	BUCKET_REVIEWS        BucketName = "reviews"
	BUCKET_DIARIES        BucketName = "diaries"
)

File: apps/api/src/lib/storage/buckets.ts
export const buckets = [
	// Add buckets here
	'default',
	'profile-images',
	'banner-images',
	'reviews',
] as const;

export const Buckets: Record<string, Bucket> = Object.fromEntries(
	buckets.map((b) => [b, b]),
);

export type Bucket = (typeof buckets)[number];
